cmd/scheduler/cmd/scheduler.go

```diff
@@ -159,25 +182,31 @@ func Run(ctx context.Context, options *scheduleroptions.Options) error {
 		logger.Error(err, "failed to create edge clientset for controller")
 		return err
 	}
-	c, err := edgeplacement.NewController(
+	es, err := scheduler.NewController(
 		kcpClusterClientset,
 		edgeClusterClientset,
-		kcpSharedInformerFactory.Scheduling().V1alpha1().Locations(),
 		edgeSharedInformerFactory.Edge().V1alpha1().EdgePlacements(),
+		kcpSharedInformerFactory.Scheduling().V1alpha1().Locations(),
+		workloadSharedInformerFactory.Workload().V1alpha1().SyncTargets(),
 	)
 	if err != nil {
-		logger.Error(err, "Failed to create controller")
+		logger.Error(err, "failed to create controller", "name", scheduler.ControllerName)
 		return err
 	}
 
 	// run the edge-scheduler
-	kubeSharedInformerFactory.Start(ctx.Done())
-	kcpSharedInformerFactory.Start(ctx.Done())
-	edgeSharedInformerFactory.Start(ctx.Done())
-	kubeSharedInformerFactory.WaitForCacheSync(ctx.Done())
-	kcpSharedInformerFactory.WaitForCacheSync(ctx.Done())
-	edgeSharedInformerFactory.WaitForCacheSync(ctx.Done())
-	c.Start(ctx, 1)
+	doneCh := ctx.Done()
+	kubeSharedInformerFactory.Start(doneCh)
+	kcpSharedInformerFactory.Start(doneCh)
+	edgeSharedInformerFactory.Start(doneCh)
+	workloadSharedInformerFactory.Start(doneCh)
+	kubeSharedInformerFactory.WaitForCacheSync(doneCh)
+	kcpSharedInformerFactory.WaitForCacheSync(doneCh)
+	edgeSharedInformerFactory.WaitForCacheSync(doneCh)
+	workloadSharedInformerFactory.WaitForCacheSync(doneCh)
+	es.Run(ctx, 1)
+	<-doneCh
```

Comment:
Why two waits on the same condition?

Reply:
Oh indeed.
Now I would like to hear your opinion, which place is better for the waiting? Here or in the controller?

Reply:
Controllers commonly have a "Run" method, although it is sometimes misnamed "Start".  I recommend following the common behavior pattern, call the method that returns when the context is closed.  Then no additional code is needed to wait for that close.

Reply:
Followed the common pattern.